python/lsst/daf/butler/_butler.py

```diff
@@ -674,26 +679,26 @@ def _findDatasetRef(self, datasetRefOrType: Union[DatasetRef, DatasetType, str],
             # If we are not searching calibration collections things may
             # fail but they are going to fail anyway because of the
             # ambiguousness of the dataId...
-            candidateDimensions = set()
+            candidateDimensions: Set[str] = set()
             candidateDimensions.update(missingDimensions)
             if datasetType.isCalibration():
                 for dim in self.registry.dimensions.getStaticDimensions():
                     if dim.temporal:
                         candidateDimensions.add(str(dim))
 
             # Look up table for the first association with a dimension
-            guessedAssociation: dict[Any, dict[str, Any]] = defaultdict(dict)
+            guessedAssociation: Dict[Any, Dict[str, Any]] = defaultdict(dict)
 
             # Keep track of whether an item is associated with multiple
             # dimensions.
-            counter = Counter()
-            assigned: dict[Any, Set[str]] = defaultdict(set)
+            counter: Any = Counter()
```

Comment:
I don't know why I assumed `typing.Counter` didn't exist, but of course it does (first I tried `Dict`, but that failed because `most_common` isn't a member, and then I got lazy).  Fixed.